agent/tests/test_ui.py

```
@@ -0,0 +1,244 @@
+        mock_instance.serve_forever.side_effect = KeyboardInterrupt
+
+        run_server(port=55199, open_browser=True)
+
+        mock_srv.assert_called_once_with(("127.0.0.1", 55199), UIHandler)
+        mock_instance.serve_forever.assert_called_once()
+        mock_instance.server_close.assert_called_once()
+
+    @patch("iconfucius.client.server.ThreadingHTTPServer")
+    @patch("iconfucius.client.server.webbrowser")
+    @patch("iconfucius.client.server._warm_cache")
+    def test_run_server_no_browser(self, mock_cache, mock_wb, mock_srv):
```

Ruff (0.15.2)

[warning] Unused method argument: `mock_cache`

(ARG002)